src/service/url-replacements-impl.js

```diff
@@ -213,8 +221,22 @@ export class UrlReplacements {
           scope: dev().assertString(scope),
           createCookieIfNotPresent: true,
         }, consent);
+      }).then(cid => {
+        if (!clientIds) {
+          clientIds = Object.create(null);
+        }
+        clientIds[scope] = cid;
+        return cid;
       });
     });
+    // Synchronous alternative. Only works for scopes that were previously
+    // requested using the async method.
+    this.set_('CLIENT_ID', scope => {
+      if (!clientIds) {
+        return null;
+      }
+      return clientIds[dev().assertString(scope)];
```

Comment:
scope is the param passed of `CLIENT_ID(someScope)`. If the scope is the same, the client ID is the same, which should be what is achieved here. The return value of this statement is `undefined` if no async request for the client id was made.